project_task_code/models/project_task.py

```diff
@@ -41,3 +41,11 @@ def name_get(self):
             name = "[{}] {}".format(rec.code, task[1])
             new_result.append((rec.id, name))
         return new_result
+
+    @api.model
+    def name_search(self, name, args=None, operator="ilike", limit=100):
+        args = args or []
+        domain = [("code", operator, name)]
```

Comment:
You need to add a check to see if the name field is not empty.
Because of this, additional values ​​are thrown up during the search